addons/settings/fnc_whitelisted.sqf

```diff
@@ -0,0 +1,39 @@
+/* ----------------------------------------------------------------------------
+Internal Function: CBA_settings_fnc_whitelisted
+
+Description:
+    Check if local machine can edit server settings.
+
+Parameters:
+    None.
+
+Returns:
+    _return - Can change server settings? <BOOL>
+
+Examples:
+    (begin example)
+        [] call CBA_settings_fnc_whitelisted
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+
+private _uid = getPlayerUID player;
+private _whitelist = getArray configFile/QGVAR(whitelist);
+
+private _cfgMissionList = missionConfigFile/QGVAR(whitelist);
+if (isArray _cfgMissionList) then {
+    _whitelist append getArray _cfgMissionList;
+};
+
+// if neither addon nor mission have white list, use wildcard for admin instead
+if (_whitelist isEqualTo []) then {
+    _whitelist = ["admin"];
+};
+
+// admin wildcard and local machine is logged in admin
+if ("admin" in _whitelist && {IS_ADMIN_LOGGED}) exitWith {true};
+
+_uid in _whitelist
```

Comment:
maybe `_whitelist` should be GVAR and set at mission init, then `FUNC(whitelisted)` become `getPlayerUID player in GVAR(whitelist)`

Reply:
no, `"admin" in GVAR(whitelist) && {IS_ADMIN_LOGGED} || {getPlayerUID player in GVAR(whitelist)}`

Reply:
But then you can't set it with addon.

Reply:
Would give the ability to dynamically add entries to the variables. Meaning admins with debug console access can just add themselves to the list. Which is exactly what the original Request wanted to prevent.

Reply:
Also true. But with the debug console, you can mess with the settings anyway if you know what you're doing.

Reply:
> But then you can't set it with addon.

why? `configFile/QGVAR(whitelist)` should exist on mission init.

Reply:
>you can mess with the settings anyway if you know what you're doing.
But not with config files.
>But then you can't set it with addon.
Think commy missunderstood. ala GVAR != config.

Reply:
Current state doesn't give any additional security because like commy2 said you can easy manage settings manually with debug console. But ability to change whitelist during mission could be helpful in some cases.